src/ansiblelint/file_utils.py

```diff
@@ -199,12 +200,60 @@ def get(self, key: Any, default: Any = None) -> Any:
 
     @property
     def content(self) -> str:
-        """Retried file content, from internal cache or disk."""
+        """Retrieve file content, from internal cache or disk."""
         if self._content is None:
             with open(self.path.resolve(), mode="r", encoding="utf-8") as f:
                 self._content = f.read()
+            if self._original_content is None:
+                self._original_content = self._content
         return self._content
 
+    @content.setter
+    def content(self, value: str) -> None:
+        """Update ``content`` and calculate ``updated``."""
+        if not isinstance(value, str):
+            raise TypeError(f"Expected str but got {type(value)}")
+        if self._original_content is None:
+            if self._content is not None:
+                self._original_content = self._content
+            elif self.path.exists():
+                # use self.content to populate self._original_content
+                _ = self.content
```

Comment:
No need to do a dummy assignment for the getattr to happen:
```suggestion
                self.content
```

Reply:
Doing that silences the pycharm linter which says the line has no effect.

Reply:
Although, that problem above is minor — a more serious problem is that an attribute access is used to perform an implicit action. This is bad — this sort of code does not carry any context, nor has the right semantic. If we see a function/method call, it's immediately obvious — some action is requested to be performed, the details can be inferred from that method name that almost certainly starts with a descriptive verb explaining that action.
One would say that if there is a need to explain what code does (not "why") with a code comment, then there is something wrong with it.
A possible improvement could be putting such an action into a private method, for example. Wouldn't it be clearer if the code looked like `self._populate_content_cache()`?

Reply:
Thank you for suggesting an interface for such a method. Sure, that makes sense.

Reply:
> Doing that silences the pycharm linter which says the line has no effect.

Which is why I suggested a follow up improvement :) It was a code smell and silencing the linter just made it shut up but not fix the underlying design issue. Although, if it was really needed, I'd opt for using a "noqa" comment with an excessive code comment explaining why the hack can't be avoided.